app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala

```diff
@@ -227,13 +262,53 @@ sealed trait State {
         Xor.left(FileWithError)
       }
     }).getOrElse(fileNotFoundError)
+  }
+
+  def sizeToByte(size: String): Long = {
+    val sizeRegex = "([1-9][0-9]{0,3})([KB,MB]{2})".r
+    size.toUpperCase match {
+      case sizeRegex(num, unit) =>
+        unit match {
+          case "KB" => num.toInt * 1024
+          case "MB" => num.toInt * 1024 * 1024
+          case _ => -1
```

Comment:
should be throwing an exception in this case, you are sending -1 instead and doing nothing with it in turn which results in returning wrong error messages (i.e. 'Envelope Size exceeds maximum' for input 'asd') for invalid constraints inputs.